sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IntelliJCacheAccessor.java

```diff
@@ -88,15 +85,18 @@ public JsonNode getDeviceCodeCredentials() throws IOException {
                 "account", "cachedAuthResult");
 
             String jsonCred  = new String(accessor.read(), StandardCharsets.UTF_8);
+
+            // If the JSON credential begins with 'cachedAuthResult@' create a substring with 'cachedAuthResult@'
+            // removed.
             if (jsonCred.startsWith("cachedAuthResult@")) {
-                jsonCred = CACHED_AUTH_RESULT_PATTERN.matcher(jsonCred).replaceFirst("");
+                jsonCred = jsonCred.substring("cachedAuthResult@".length());
```

Comment:
Replace `Pattern` usage with substring as we check if the string begins with the value that we regex replaceFirst, effectively a substring operation